go/service/log.go

```diff
@@ -4,45 +4,40 @@
 package service
 
 import (
-	"fmt"
-
 	"golang.org/x/net/context"
 
+	"github.com/keybase/client/go/libkb"
 	keybase1 "github.com/keybase/client/go/protocol"
+	rpc "github.com/keybase/go-framed-msgpack-rpc"
 )
 
-type LogUI struct {
-	sessionID int
-	cli       *keybase1.LogUiClient
+// LogHandler is the RPC handler for the log interface.
+type LogHandler struct {
+	*BaseHandler
+	logReg *logRegister
+	libkb.Contextified
 }
 
-func (l *LogUI) Log(level keybase1.LogLevel, format string, args []interface{}) {
-	msg := fmt.Sprintf(format, args...)
-	l.cli.Log(context.TODO(), keybase1.LogArg{
-		SessionID: l.sessionID,
-		Level:     keybase1.LogLevel(level),
-		Text: keybase1.Text{
-			Markup: false,
-			Data:   msg,
-		},
-	})
+// NewLogHandler creates a LogHandler for the xp transport.
+func NewLogHandler(xp rpc.Transporter, logReg *logRegister, g *libkb.GlobalContext) *LogHandler {
+	return &LogHandler{
+		BaseHandler:  NewBaseHandler(xp),
+		logReg:       logReg,
+		Contextified: libkb.NewContextified(g),
+	}
 }
 
-func (l *LogUI) Debug(format string, args ...interface{}) {
-	l.Log(keybase1.LogLevel_DEBUG, format, args)
-}
-func (l *LogUI) Info(format string, args ...interface{}) {
-	l.Log(keybase1.LogLevel_INFO, format, args)
-}
-func (l *LogUI) Critical(format string, args ...interface{}) {
-	l.Log(keybase1.LogLevel_CRITICAL, format, args)
-}
-func (l *LogUI) Warning(format string, args ...interface{}) {
-	l.Log(keybase1.LogLevel_WARN, format, args)
-}
-func (l *LogUI) Errorf(format string, args ...interface{}) {
-	l.Log(keybase1.LogLevel_ERROR, format, args)
-}
-func (l *LogUI) Notice(format string, args ...interface{}) {
-	l.Log(keybase1.LogLevel_NOTICE, format, args)
+func (h *LogHandler) RegisterLogger(_ context.Context, arg keybase1.RegisterLoggerArg) (err error) {
+	h.G().Log.Debug("LogHandler::RegisterLogger: %+v", arg)
```

Comment:
given the issues we've had with log forwarding in the past, I thought it might be nice to have both.